tests/lib/config-array-factory.js

```diff
@@ -38,6 +42,22 @@ const eslintAllPath = path.resolve(dirname, "../fixtures/eslint-all.cjs");
 const eslintRecommendedPath = path.resolve(dirname, "../fixtures/eslint-recommended.cjs");
 const tempDir = path.join(systemTempDir, "eslintrc/config-array-factory");
 
+/**
+ * Return config data for built-in eslint:all.
+ * @returns {ConfigData} Config data
+ */
+function getEslintAllConfig() {
+    return import("../fixtures/eslint-all.cjs");
+}
+
+/**
+ * Return config data for built-in eslint:recommended.
+ * @returns {ConfigData} Config data
+ */
+function getEslintRecommendedConfig() {
+    return import("../fixtures/eslint-recommended.cjs");
+}
+
```

Comment:
This doesn't work as expected, because `import()` returns a Promise. We can probably use `require()` here (`require` is created on line 24).

I believe that's the reason why one test is failing:

```
635 passing (2s)
  1 failing

  1) ConfigArrayFactory
       '_normalizeConfigData(configData, ctx)' method should normalize the config data.
         'extends' details
           if 'extends' property was 'eslint:all', the returned value
             should have the config data of 'eslint:all' at the first element.:

      AssertionError: expected { type: 'config', …(15) } to deeply equal { …(16) }
      + expected - actual

       {
         "criteria": [null]
         "env": [undefined]
      -  "filePath": ""
      +  "filePath": "/home/runner/work/eslintrc/eslintrc/tests/fixtures/eslint-all.cjs"
         "globals": [undefined]
         "ignorePattern": [undefined]
         "name": ".eslintrc » eslint:all"
         "noInlineConfig": [undefined]
--
         "plugins": [undefined]
         "processor": [undefined]
         "reportUnusedDisableDirectives": [undefined]
         "root": [undefined]
      -  "rules": [undefined]
      +  "rules": {
      +    "curly": [
      +      "error"
      +    ]
      +    "eqeqeq": [
      +      "error"
      +    ]
      +  }
         "settings": [undefined]
         "type": "config"
       }
      
      at assertConfigArrayElement (file:///home/runner/work/eslintrc/eslintrc/tests/lib/config-array-factory.js:[88](https://github.com/eslint/eslintrc/runs/5220072307?check_suite_focus=true#step:5:88):12)
      at Context.<anonymous> (file:///home/runner/work/eslintrc/eslintrc/tests/lib/config-array-factory.js:[104](https://github.com/eslint/eslintrc/runs/5220072307?check_suite_focus=true#step:5:104)4:21)
```

You can run `npm test` locally to check.

What's surprising is that only one test is failing. Can we, instead of semi-randomly mixing strings and functions in tests, convert all tests to use the new functions `getEslintAllConfig` and `getEslintRecommendedConfig`, and then duplicate tests that have `eslint:all` and `eslint:recommended` to check if they work with `eslintAllPath` and `eslintRecommendedPath`, for backwards compatibility.

Reply:
Sorry for missing the failed tests. I'll fix it.

Reply:
Sorry for missing the failed tests. Will fix it.